provider/server/exec_test.go

```diff
@@ -38,12 +38,14 @@ var _ = Describe("Exec", func() {
 		Expect(createResp).NotTo(BeNil())
 
 		DeferCleanup(func(ctx SpecContext) {
-			Eventually(func(g Gomega) bool {
+			Eventually(func(g Gomega) {
```

Comment:
I don't think it is a good idea to call `DeleteMachine` in an `Eventually` block. We should cal the delete method once and then ensure that the domain has been removed.

Reply:
@afritzler do you understand why i want to have Eventually block for delete call? Can i close your comment?